scripts/install/install-dependencies.sh

```diff
@@ -253,223 +587,288 @@ function install_latest_cfitsio() {
+orig_panoptes_env_checksum="$(checksum_panoptes_env)"
 
-if [[ "${DO_CONDA}" -eq 1 ]] ; then
-  install_conda_if_missing
+# Before installing conda, figure out if the calling shell had the correct
+# environment setup.
+had_activated_panoptes_env=0
+if is_panoptes_env_activated ; then
+  had_activated_panoptes_env=1
 fi
 
-
-# Make sure we use the correct Anaconda environment.
-DO_CREATE_CONDA_ENV=0
-if [[ -z "$(conda info --envs | grep panoptes-env)" ]] ; then
-  DO_CREATE_CONDA_ENV=1
-elif [[ "${DO_REBUILD_CONDA_ENV}" -eq 1 ]] ; then
-  source activate root
-  conda remove --all --yes --quiet -n panoptes-env
-  DO_CREATE_CONDA_ENV=1
-fi
-if [[ "${DO_CREATE_CONDA_ENV}" -eq 1 ]] ; then
-  conda create --yes -n panoptes-env python=3
+# Install Conda, a Python package manager from Anaconda, Inc. Supports both
+# pure Python packages (just as pip does) and packages with non-Python parts
+# (e.g. native libraries).
+if [[ "${DO_CONDA}" -eq 1 ]] ; then
+  maybe_install_conda
 fi
 
-add_to_profile "source activate panoptes-env"
-source activate panoptes-env
+if [[ "${DO_CONDA}" -eq 1 || \
+      "${DO_REBUILD_CONDA_ENV}" -eq 1 || \
+      "${DO_INSTALL_CONDA_PACKAGES}" -eq 1 ]] ; then
+  prepare_panoptes_conda_env
+fi
 
-if [[ "${DO_CONDA}" -eq 1 || "${DO_CREATE_CONDA_ENV}" -eq 1 || \
-      "${DO_PIP_REQUIREMENTS}" -eq 1 ]] ; then
+if [[ -z "$(safe_type conda)" ]] ; then
   echo_bar
-  echo
-  echo "Updating conda installation."
-  conda update --quiet --all --yes
+  echo "
+Error: conda is not installed, but remaining installation steps make
+use of the conda environment for PANOPTES (panoptes-env).
+"
+  exit 1
 fi
 
+# Activate the PANOPTES environment in this shell.
+conda activate panoptes-env
 
-if [[ "${DO_INSTALL_CONDA_PACKAGES}" -eq 1 ]] ; then
+# Install pure Python packages using pip; note that we prefer to
+# install these with conda if they are available that way, as we
+# try to reduce the number of package managers and repositories
+# we have to deal with.
+if [[ "${DO_PIP_REQUIREMENTS}" -eq 1 ]] ; then
   echo_bar
   echo
-  echo "Installing conda packages"
-  conda install --yes "--file=${THIS_DIR}/conda-packages-list.txt"
-fi
-
+  echo "Upgrading pip before installing other python packages."
+  pip install --quiet --upgrade pip
 
-if [[ "${DO_PIP_REQUIREMENTS}" -eq 1 ]] ; then
-  # Upgrade pip itself before installing other python packages.
-  pip install -U pip
-  # TODO(jamessynge): Move this script and needed text files into an install
-  # directory.
-  # TODO(wtgee): Consider whether to inline the needed text files into this
-  # file, and add git clone of the PANOPTES repos, so that the user can do
-  # an install with just about two commands (wget script, then run script).
-  pip install -r "${POCS}/requirements.txt"
+  echo_bar
+  echo
+  echo "Installing python packages using pip."
+  pip install --quiet --requirement "${POCS}/requirements.txt"
 fi
 
-
+# Install cfitsio, native tools for reading and writing FITS files.
```

Comment:
Did cfitsio move to a conda package and if so does it need a separate option for installing? Would we ever not install cfitsio?